python/src/dataset.rs

```diff
@@ -275,39 +278,72 @@ impl Dataset {
         kwargs: Option<&PyDict>,
     ) -> PyResult<()> {
         let idx_type = match index_type.to_uppercase().as_str() {
-            "IVF_PQ" => IndexType::Vector,
+            "IVF_PQ" | "DISKANN" => IndexType::Vector,
             _ => {
                 return Err(PyValueError::new_err(format!(
                     "Index type '{index_type}' is not supported."
                 )))
             }
         };
 
-        // Only VectorParams are supported.
-        let mut params = VectorIndexParams::default();
-        if let Some(kwargs) = kwargs {
-            if let Some(n) = kwargs.get_item("num_partitions") {
-                params.num_partitions = PyAny::downcast::<PyInt>(n)?.extract()?
-            };
-
-            if let Some(n) = kwargs.get_item("num_sub_vectors") {
-                params.num_sub_vectors = PyAny::downcast::<PyInt>(n)?.extract()?
-            };
-
-            if let Some(o) = kwargs.get_item("use_opq") {
-                params.use_opq = PyAny::downcast::<PyBool>(o)?.extract()?
-            };
-            if let Some(o) = kwargs.get_item("max_opq_iterations") {
-                params.max_opq_iterations = PyAny::downcast::<PyInt>(o)?.extract()?
-            };
-        }
-
-        params.metric_type = match metric_type {
+        let m_type = match metric_type {
             Some(mt) => MetricType::try_from(mt.to_string().to_lowercase().as_str())
                 .map_err(|err| PyValueError::new_err(err.to_string()))?,
             None => MetricType::L2,
         };
 
+        // Only VectorParams are supported.
+        let params = match index_type.to_uppercase().as_str() {
+            "IVF_PQ" => {
+                let mut ivf_params = IvfBuildParams::default();
+                let mut pq_params = PQBuildParams::default();
+                if let Some(kwargs) = kwargs {
+                    if let Some(n) = kwargs.get_item("num_partitions") {
+                        ivf_params.num_partitions = PyAny::downcast::<PyInt>(n)?.extract()?
+                    };
+
+                    if let Some(n) = kwargs.get_item("num_bits") {
```

Comment:
i forgot do we actually support configuring this now?

Reply:
We dont support "throw error for unsupported params". This is just do the handing silently. I am fine to remove it if you feel strongly.